src/Designer/backend/src/Designer/Middleware/UserRequestSynchronization/OrgWide/RequestSyncEvaluators/EndpointNameSyncEligibilityEvaluator.cs

```diff
@@ -0,0 +1,54 @@
+using System.Collections.Frozen;
+using System.Collections.Generic;
+using Altinn.Studio.Designer.Controllers;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Controllers;
+
+namespace Altinn.Studio.Designer.Middleware.UserRequestSynchronization.OrgWide.RequestSyncEvaluators;
+
+/// <summary>
+/// Evaluates if a request is eligible for synchronization based on the endpoint name.
+/// Contains a whitelist of endpoints that are eligible for synchronization.
+/// </summary>
+public class EndpointNameSyncEligibilityEvaluator : IOrgWideSyncEligibilityEvaluator
+{
+    private const string RemoveControllerSuffix = "Controller";
+    private static string TrimmedControllerName(string controllerName) => controllerName.Replace(RemoveControllerSuffix, string.Empty);
+    private readonly FrozenDictionary<string, FrozenSet<string>> _endpointsWhiteList = new Dictionary<string, FrozenSet<string>>
+    {
+        {
+            TrimmedControllerName(nameof(DeploymentsController)),
+            GenerateFrozenSet(
+                nameof(DeploymentsController.Create),
+                nameof(DeploymentsController.Undeploy)
+            )
+        }
+    }.ToFrozenDictionary();
+
+    private static FrozenSet<string> GenerateFrozenSet(params string[] actions)
+    {
+        return actions.ToFrozenSet();
+    }
+
+    public bool IsEligibleForSynchronization(HttpContext httpContext)
+    {
+        var endpoint = httpContext.GetEndpoint();
+
+        var controllerActionDescriptor = endpoint?.Metadata.GetMetadata<ControllerActionDescriptor>();
+
+        if (controllerActionDescriptor == null)
```

Comment:
Should we perhaps throw an exception for cases like his? If we ever were to migrate to minimal APIs we would probably have to go digging/debugging to figure why things aren't working as expected? Exceptions tend to be more obvious 😄